contracts/Aavegotchi/facets/AavegotchiLendingFacet.sol

```diff
@@ -0,0 +1,291 @@
+        require(_revenueSplit[0] + _revenueSplit[1] + _revenueSplit[2] == 100, "AavegotchiLending: sum of revenue split should be 100");
+        if (_receiver == address(0)) {
+            require(_revenueSplit[2] == 0, "AavegotchiLending: revenue split for invalid receiver should be zero");
+        }
+        require(_whitelistId == 0 || s.whitelistOwners[_whitelistId] == sender, "AavegotchiLending: Not owner of whitelist");
+
+        require(s.aavegotchis[_erc721TokenId].status == LibAavegotchi.STATUS_AAVEGOTCHI, "AavegotchiLending: Only aavegotchi available");
+
+        uint256 oldRentalId = s.aavegotchiRentalHead[_erc721TokenId];
+        if (oldRentalId != 0) {
+            LibAavegotchiLending.cancelAavegotchiRental(oldRentalId, sender);
+        } else {
+            require(s.aavegotchis[_erc721TokenId].locked == false, "AavegotchiLending: Only callable on unlocked Aavegotchis");
+        }
+
+        s.nextAavegotchiRentalId++;
+        uint256 rentalId = s.nextAavegotchiRentalId;
+
+        s.aavegotchiRentalHead[_erc721TokenId] = rentalId;
+        s.aavegotchiRentals[rentalId] = AavegotchiRental({
+            rentalId: rentalId,
+            amountPerDay: _amountPerDay,
+            period: _period,
+            revenueSplit: _revenueSplit,
+            originalOwner: sender,
+            renter: address(0),
+            receiver: _receiver,
+            erc721TokenAddress: _erc721TokenAddress,
+            erc721TokenId: _erc721TokenId,
+            whitelistId: _whitelistId,
+            timeCreated: block.timestamp,
+            timeAgreed: 0,
+            lastClaimed: 0,
+            canceled: false,
+            completed: false
+        });
+
+        emit AavegotchiRentalAdd(rentalId, sender, _erc721TokenAddress, _erc721TokenId, _amountPerDay, _period, block.timestamp);
+
+        // Lock Aavegotchis when listing is created
+        if (_erc721TokenAddress == address(this)) {
+            s.aavegotchis[_erc721TokenId].locked = true;
+        }
+    }
+
+    ///@notice Allow an original aavegotchi owner to cancel his NFT rental by providing the NFT contract address and identifier
+    ///@param _erc721TokenId The identifier of the NFT to be delisted from rental
+    function cancelAavegotchiRentalByToken(uint256 _erc721TokenId) external {
+        LibAavegotchiLending.cancelAavegotchiRentalFromToken(_erc721TokenId, LibMeta.msgSender());
+    }
+
+    ///@notice Allow an original aavegotchi owner to cancel his NFT rental through the listingID
+    ///@param _rentalId The identifier of the rental to be cancelled
+    function cancelAavegotchiRental(uint256 _rentalId) external {
+        LibAavegotchiLending.cancelAavegotchiRental(_rentalId, LibMeta.msgSender());
+    }
+
+    ///@notice Allow a renter to agree an rental for the NFT
+    ///@dev Will throw if the NFT has been lent or if the rental has been canceled already
+    ///@param _rentalId The identifier of the rental to agree
+    function agreeAavegotchiRental(
```

Comment:
An owner can list an Aavegotchi for rent.   Then a renter can rent that Aavegotchi and then the renter can put the Aavegotchi for sale in the Baazaar and sell it there.  Locked Aavegotchis don't prevent them from being listing/sold in the ERC721 marketplace facet.

Reply:
Great catch. We should prevent locked Aavegotchis from being listed in the baazaar.